public/app/features/query/state/PanelQueryRunner.ts

```diff
@@ -219,7 +219,8 @@ export class PanelQueryRunner {
   private applyTransformations(data: PanelData): Observable<PanelData> {
     const transformations = this.dataConfigSource.getTransformations();
 
-    if (!transformations || transformations.length === 0) {
+    const allTransformationsDisabled = transformations && transformations.every((t) => t.disabled);
+    if (allTransformationsDisabled || !transformations || transformations.length === 0) {
       return of(data);
     }
 
```

Comment:
This is the actual bug fix, the rest of the changes are e2e test related 😅 

For context this is just ejecting out of applying transformations if we detect that all passed transforms are disabled which is a slight optimization.

The main bug lies down on [lines 234-237](https://github.com/grafana/grafana/pull/81320/files#diff-0578c99d69d9e4905d3eac87e12f2eafd5fd9e0d4d05d1735c85c26bcc7f9cb5R234-R237) and is a tricky rxjs edge case that seems to only occur when there is only 1 disabled transform 🤷‍♂️

Reply:
I also don't **_think_** this will break anything 😅 👍